packages/react/src/ReactElementValidator.js

```diff
@@ -134,6 +134,7 @@ function validateExplicitKey(element, parentType) {
   // assigning it a key.
   let childOwner = '';
   if (
+    __DEV__ &&
```

Comment:
Last place where `element._owner` is read I believe. Responsible for enhancing "should have unique `key` prop":  

```jsx
function InnerComponent({ childSet }) {
  return <div>{childSet}</div>;
}

class App extends Component {
  render() {
    return <InnerComponent childSet={[<div />, <div />]} />;
  }
}
```

```bash
Warning: Each child in a list should have a unique "key" prop.

Check the render method of `InnerComponent`. It was passed a child from App. See https://reactjs.org/link/warning-keys for more information.
    at div
    at InnerComponent (https://cggv72.csb.app/src/index.js:14:3)
    at App (https://cggv72.csb.app/src/index.js:23:1)
```
-- https://codesandbox.io/s/warnings-with-owner-usage-cggv72

Reply:
this whole file is / should be dev-only